agents/mcts.c

```diff
@@ -0,0 +1,158 @@
+#include <assert.h>
+#include <float.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "game.h"
+#include "mcts.h"
+#include "util.h"
+
+struct node {
+    int move;
+    char player;
+    int n_visits;
+    double score;
+    struct node *parent;
+    struct node *children[N_GRIDS];
+};
+
+static struct node *new_node(int move, char player, struct node *parent)
+{
+    struct node *node = malloc(sizeof(struct node));
+    node->move = move;
+    node->player = player;
+    node->n_visits = 0;
+    node->score = 0;
+    node->parent = parent;
+    memset(node->children, 0, sizeof(node->children));
+    return node;
+}
+
+static void free_node(struct node *node)
+{
+    for (int i = 0; i < N_GRIDS; i++)
+        if (node->children[i])
+            free_node(node->children[i]);
+    free(node);
+}
+
+static inline double uct_score(int n_total, int n_visits, double score)
+{
+    if (n_visits == 0)
+        return DBL_MAX;
+    return score / n_visits + C * sqrt(log(n_total) / n_visits);
+}
+
+static struct node *select_move(struct node *node)
+{
+    struct node *best_node = NULL;
+    double best_score = -1;
+    for (int i = 0; i < N_GRIDS; i++) {
+        if (node->children[i]) {
```

Comment:
Rewrite as following:
```c
    if (!node->children[i])
        continue;
    double score = uct_score(node->n_visits, node->children[i]->n_visits, ...
```

So that, we can reduce the indentions.